File: src/SynthShop.UI/src/app/models/basket.ts
import { Product } from "./product"

export interface Basket {
    basketId: string
    CustomerId: string | null
    items: BasketItem[]
  }
  
  export interface BasketItem {
    basketItemId: string
    quantity: number
    productId: string
    product: Product
    basketId: string
  } 

File: src/SynthShop.Domain/Entities/Basket.cs
namespace SynthShop.Domain.Entities;

public class Basket
{
    public Guid BasketId { get; set; }
    public Guid? CustomerId { get; set; }

    public List<BasketItem> Items { get; set; }

    public Basket()
    {
        Items = new List<BasketItem>();
    }
}